### PR TITLE
Disallow too-coarse-grained cycle point formats

### LINKED ISSUE #1449: Fail validation if cycle point format is too coarse-grained.
If the cycle point format is coarser-grained than a recurrence, task cycle points (in effect) don't increment.  This is kind of obvious, but I've done it myself accidentally several times.  We should probably fail it in validation.

```
[cylc]
    cycle point format = %Y-%m
[scheduling]
    initial cycle point = 2015-08
    [[dependencies]]
        [[[P1D]]]
            graph = foo
```
This results in a very short suite run (see the WARNING):
```
2015-07-27T12:30:54+12 INFO - port:7766
2015-07-27T12:30:54+12 INFO - Suite starting at 2015-07-27T12:30:54+12
2015-07-27T12:30:54+12 INFO - Run mode: live
2015-07-27T12:30:54+12 INFO - Initial point: 2015-08
2015-07-27T12:30:54+12 INFO - Final point: None
2015-07-27T12:30:54+12 INFO - Cold Start 2015-08
2015-07-27T12:30:54+12 INFO - [foo.2015-08] -job(01) initiate job-submit
2015-07-27T12:30:55+12 INFO - [foo.2015-08] -submit_method_id=13233
2015-07-27T12:30:55+12 INFO - [foo.2015-08] -submission succeeded
2015-07-27T12:30:55+12 WARNING - foo.2015-08 cannot be added to pool: task ID already exists
2015-07-27T12:30:55+12 INFO - [foo.2015-08] -(current:submitted)> foo.2015-08 started at 2015-07-27T12:30:54+12
2015-07-27T12:30:58+12 INFO - [foo.2015-08] -(current:running)> foo.2015-08 succeeded at 2015-07-27T12:30:57+12
2015-07-27T12:30:59+12 INFO - Suite shutting down at 2015-07-27T12:30:59+12
```

[UPDATE: corrected for typo in the original description]

### PR DESCRIPTION
**Pull request by benfitzpatrick**
Closes #1449.

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
@hjoliver, please review 1.

**Comment by hjoliver** [comment]
Looks good, tests as working.  @matthewrmshin - please review 2 or reassign.

File: tests/validate/39-degenerate-point-format.t [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Test validation with a new-style cycle point and an async graph.
+. $(dirname $0)/test_header
+#-------------------------------------------------------------------------------
+set_test_number 2
+#-------------------------------------------------------------------------------
+install_suite $TEST_NAME_BASE $TEST_NAME_BASE
+#-------------------------------------------------------------------------------
+TEST_NAME=$TEST_NAME_BASE
+run_fail $TEST_NAME cylc validate --debug -v -v $SUITE_NAME
+grep_ok "Sequence R/2015-08/P1D: equal adjacent points: 2015-08 => 2015-08." $TEST_NAME.stderr
```
Comment (matthewrmshin) [inline tests/validate/39-degenerate-point-format.t]: May be worth adding the date time format in the error message to make it more obvious to users?

File: tests/validate/39-degenerate-point-format.t [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#-------------------------------------------------------------------------------
+exit
```
Comment (matthewrmshin) [inline tests/validate/39-degenerate-point-format.t]: Do we need to clean up suite here?

**Comment by benfitzpatrick** [comment]
Added the cycle point format to the error message.

A call to `purge_suite` isn't needed here, as the suite is installed in a
temporary (`/var/tmp/`) directory and never run.

**Pull request merged**